server/src/packages/pas-form/pas-form.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { CrmService } from '../../crm/crm.service';
+import { PACKAGE_ATTRS } from '../packages.controller';
+import { PAS_FORM_ATTRS } from './pas-form.controller';
+import { PACKAGE_TYPE_OPTIONSET } from '../packages.service';
+
+@Injectable()
+export class PasFormService {
+  constructor(private readonly crmService: CrmService) {}
+
+  async find(id) {
+    const { records: [pasForm] } = await this.crmService.get(`dcp_pasforms`, `
+      $select=${PAS_FORM_ATTRS.join(',')}
+      &$filter=
+        dcp_pasformid eq ${id}
+      &$expand=
+        dcp_dcp_applicantinformation_dcp_pasform,
+        dcp_dcp_applicantrepinformation_dcp_pasform,
+        dcp_package($select=${PACKAGE_ATTRS.join(',')}),
+        dcp_dcp_projectbbl_dcp_pasform($filter=statecode eq 0),
+        dcp_dcp_projectaddress_dcp_pasform
+    `);
+
+    // CRM feature for providing the human-readable value for the related resource
+    const { '_dcp_projectname_value@OData.Community.Display.V1.FormattedValue': dcp_projectname } = pasForm;
+
+    return {
+      ...pasForm,
+
+      dcp_revisedprojectname: pasForm.dcp_revisedprojectname || dcp_projectname,
+    };
+  }
+
+  // Gets the lateset PAS Package on Project `projectId`
+  async getLatestPasForm(projectId): Promise<any> {
+    try {
+      const { records: pasPackages } = await this.crmService.get('dcp_packages', `
```

Comment:
Just curious/for discussion: should this function go int he Package service, since it acts on dcp_packages?
Or should it stay here in the pas-form service, since it returns a PAS Form?

Reply:
That's a good question — it does call `dcp_packages`, but it's ultimately supposed to return a PAS-FORM. It takes a project ID as an argument, which is a little confusing as well. Maybe the argument should be changed?

I find that querying `dcp_packages` is just a detail of the implementation. We want to create a concise abstraction, concealing the nitty-gritty underneath.